### PR TITLE
Extend and re-arrange documentation

### PR DESCRIPTION
**Pull request by mkorszun**
* Introduce guides explaining basic consumer patterns and configurations
* Split telemetry events by consumer / publisher and move to separate files
* Move `guides` and `examples` to the `documentation` folder

## Checklist

- [ ] I have added unit tests to cover my changes.
- [ ] I have improved the code quality of this repo. (refactoring, or reduced number of static analyser issues)
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: guides/consumer/basic_setup.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Consumer basic setup
+====================
+
+# Example
```
Comment (spier) [inline guides/consumer/basic_setup.md:4]: What is the difference between these two different ways to indicate a headline?

File: lib/consumer.ex [hunk 2]
```diff
@@ -99,7 +99,8 @@ defmodule GenRMQ.Consumer do
         durable: true,
         passive: true,
         arguments: [
-          {"x-queue-type", :longstr ,"quorum"},
+          {"x-queue-type", :longstr ,"quorum"}
+        ]
```
Comment (spier) [inline lib/consumer.ex:103]: Is this change meant to be in this PR?

File: lib/consumer.ex [hunk 3]
```diff
@@ -113,7 +114,8 @@ defmodule GenRMQ.Consumer do
       deadletter_queue: "gen_rmq_in_queue_error",
       deadletter_queue_options: [
         arguments: [
-          {"x-queue-type", :longstr ,"quorum"},
+          {"x-queue-type", :longstr ,"quorum"}
+        ]
```
Comment (spier) [inline lib/consumer.ex:118]: Is this change meant to be in this PR?

File: lib/consumer.ex (hunk 2 above)
Comment (mkorszun) [reply lib/consumer.ex:103]: Yes, this are the code docs and there was a typo.

File: lib/consumer.ex (hunk 3 above)
Comment (mkorszun) [reply lib/consumer.ex:118]: Yes, it is a code documentation. There was a typo.

File: guides/consumer/basic_setup.md (hunk 1 above)
Comment (mkorszun) [reply guides/consumer/basic_setup.md:4]: Very good question. I do not know. I always use `===` for the title and then `#` for headlines

**Comment by mkorszun** [comment]
@spier 
> Questions:
> 
>     * different headlines used
>     * `quorum` references

* I can change headline :)
* What about `quorum`?

**Comment by spier** [comment]
I understand the `quorum` thing now.

What do you think about moving guides and examples into a folder `docs`?
i.e.  we would have:

* `docs/examples`
* `docs/guides`

**Comment by mkorszun** [comment]
@spier `doc` is already predefined output for auto-generated docs required to publish a package to hex. So I would rather move everything under `documentation`

**Comment by mkorszun** [comment]
@spier moved everything under `documentation`

**Comment by mkorszun** [comment]
@spier let me know your final thoughts here.

**Pull request merged**